Assets/js/05.budget_widget.js

```diff
@@ -0,0 +1,99 @@
+            mQuery.ajax({
+                url: mauticAjaxUrl,
+                type: 'POST',
+                data: {
+                    action: 'plugin:mauticContactSource:campaignBudgetsDashboard',
+                },
+                cache: true,
+                dataType: 'json',
+                success: function (response) {
+                    var rowCount = Math.floor((widgetHeight - 140) / 40);
+                    mQuery('#budgets-widget').DataTable({
+                        language: {
+                            emptyTable: 'No results found for this date range and filters.'
+                        },
+                        data: response.rows,
+                        autoFill: true,
+                        columns: response.columns,
+                        order: [[2, 'asc'], [4, 'asc']],
+                        bLengthChange: false,
+                        lengthMenu: [[rowCount]],
+                        dom: '<<lBf>rtip>',
+                        buttons: [
+                            'excelHtml5',
+                            'csvHtml5'
+                        ],
+                        columnDefs: [
+                            {
+                                render: function (data, type, row) {
+                                    return renderStatusToggle(row[0]);
+                                },
+                                targets: 0
+                            },
+                            {
+                                // campaign name and link
+                                render: function (data, type, row) {
+                                    return renderLink(row[1], row[2]);
+                                },
+                                targets: 1
+                            },
+                            {
+                                // source name and link
+                                render: function (data, type, row) {
+                                    return renderLink(row[3], row[4]);
+                                },
+                                targets: 3
+                            },
+                            {
+                                // forecast
+                                render: function (data, type, row) {
+                                    return renderForecast(row[8], row[9]);
+                                },
+                                targets: 8
+                            },
+                            {visible: false, targets: [2, 4, 9]},
+                            {width: '5%', targets: [0]},
+                            {width: '45%', targets: [5]},
+                        ]
+                    });
+                    mQuery('#budget-widget-overlay').hide();
+                    mQuery('#budgets-widget_wrapper .dt-buttons').css({float: "right", marginLeft: "10px"});
+                }
+            });
+        });
+    }
+};
+
+function renderStatusToggle (icon) {
```

Comment:
Can we move these 3 functions inside the parent above, or just make them callbacks?

Reply:
It Shall be done